stellargraph/core/graph.py

```diff
@@ -751,6 +751,18 @@ def node_type(self, node, use_ilocs=False):
         assert len(type_sequence) == 1
         return type_sequence[0]
 
+    def vectorized_node_type(self, node_ilocs):
```

Comment:
What about extending `node_type` to work on an array?

Reply:
I felt like this was cleaner compared to doubling the code paths in `node_type` again, especially for something that'll be mostly a fast internal method. What do you think?

Reply:
I think it's a relatively controlled doubling, because it can likely be done by conditionalising the `nodes = [node]` and `assert ... [0]`? In addition, the function is tiny.

My concern is that it's a clunky name (e.g. it's long, we don't use "vectorised" anywhere else in our API, and doesn't sort next to the other `node_type` methods; the last one can be addressed by `node_type_vectorized`) and that I could easily see a user using this directly, since bulk queries are always great.